### ISSUE TITLE
Duplicate strings in en.php

### ISSUE THREAD

**Issue by CaraesNaur** [flyspray/flyspray#908]
While working on #892 I stumbled into `en.php` contains 29 duplicate strings.  Some are false positives (such as the text for severity and priority levels).

Below is the output of a script that audits `en.php` for duplicates.  Each header is a string value.  The lists show each entry's line number and content in the file.  False positives are notes as ~~strikethrough~~.

This is on the current master branch `HEAD`.  A further audit of their usage throughout the codebase will be needed before deciding whether any can be combined, although some are pretty obvious.

Perhaps there are reasons for some of them that aren't apparent.

### "All"

 * `803:'All'                     => 'All',`
 * `1137:'showaccountsall'         => 'All',`

### "Attach a file"

 * `234:'uploadafile'             => 'Attach a file',`
 * `595:'attachafile'             => 'Attach a file',`

### "Can view roadmap"

 * `998:'canviewroadmap'          => 'Can view roadmap',`
 * `1021:'viewroadmap'             => 'Can view roadmap',`

### "Create a new Project"

 * `100:'createproject'           => 'Create a new Project',`
 * `585:'createnewproject'        => 'Create a new Project',`

### "Date"

 * `268:'date'                    => 'Date',`
 * `273:'eventdate'               => 'Date',`

### "Delete this item from the list"

 * `186:'listdeletetip'           => 'Delete this item from the list',`
 * `861:'deletetip'               => 'Delete this item from the list',`

### "Dependency added"

 * `313:'depadded'                => 'Dependency added',`
 * `731:'pm.depadded'             => 'Dependency added',`

### "Dependency removed"

 * `315:'depremoved'              => 'Dependency removed',`
 * `732:'pm.depremoved'           => 'Dependency removed',`

### "Event log"

 * `437:'reports'                 => 'Event log',`
 * `800:'eventlog'                => 'Event log',`

### ~~"High"~~

 * `654:'priority5'               => 'High',`
 * `702:'severity4'               => 'High',`

### ~~"Low"~~

 * `656:'priority3'               => 'Low',`
 * `704:'severity2'               => 'Low',`

### ~~"Medium"~~

 * `655:'priority4'               => 'Medium',`
 * `703:'severity3'               => 'Medium',`

### "Members can login"

 * `37:'groupenabled'            => 'Members can login',`
 * `38:'groupopen'               => 'Members can login',`

### "Opened"

 * `372:'dateopened'              => 'Opened',`
 * `681:'opened'                  => 'Opened',`

### "Operating System"

 * `213:'operatingsystem'         => 'Operating System',`
 * `379:'os'                      => 'Operating System',`

### "Reason for denial"

 * `627:'denialreason'            => 'Reason for denial',`
 * `850:'reasonfordeinal'         => 'Reason for denial',`

### "Show image attachments inline"

 * `107:'showinlineimages'        => 'Show image attachments inline',`
 * `589:'inlineimages'            => 'Show image attachments inline',`

### "Show this item in the list"

 * `103:'listshowtip'             => 'Show this item in the list',`
 * `860:'showtip'                 => 'Show this item in the list',`

### "Task reopened"

 * `278:'taskreopened'            => 'Task reopened',`
 * `730:'pm.taskreopened'         => 'Task reopened',`

### "That user does not exist on this Flyspray installation."

 * `816:'error5'                  => 'That user does not exist on this Flyspray installation.',`
 * `833:'error19'                 => 'That user does not exist on this Flyspray installation.',`

### "The order these items will appear in the list"

 * `102:'listordertip'            => 'The order these items will appear in the list',`
 * `859:'ordertip'                => 'The order these items will appear in the list',`

### "The task depends upon"

 * `308:'taskdependson'           => 'The task depends upon',`
 * `309:'taskdependsontask'       => 'The task depends upon',`
 * `310:'taskdependsontasks'      => 'The task depends upon',`

### ~~"Very Low"~~

 * `657:'priority2'               => 'Very Low',`
 * `705:'severity1'               => 'Very Low',`

### "Vote"

 * `36:'vote'                    => 'Vote',`
 * `412:'tovote'                  => 'Vote',`

### "hour"

 * `257:'hour'                    => 'hour',`
 * `993:'hoursingular'            => 'hour',`

### "hours"

 * `258:'hours'                   => 'hours',`
 * `992:'hourplural'              => 'hours',`

### "minute"

 * `987:'minute'                  => 'minute',`
 * `990:'minutesingular'          => 'minute',`

### "minutes"

 * `988:'minutes'                 => 'minutes',`
 * `989:'minuteplural'            => 'minutes',`

### "to"

 * `671:'to'                      => 'to',`
 * `775:'selectduedateto'         => 'to',`
 * `777:'selectsincedateto'       => 'to',`
 * `780:'selectopenedto'          => 'to',`
 * `782:'selectclosedto'          => 'to',`

**Comment by peterdd** [comment]
Yes, having different placeholders that result in the same translation in english might have differing strings in other languages, depending on the context.

The reuse of a translation identifier must be checked in context.

So better have some duplicated english strings with verbose unambiguous identifiers.

There might cases where some translation identifiers could be merged into one, but every usage and context in the code must be checked first. Just comparing the en.php is not enough there. Also existing translations in other languages must be compared then.

**Comment by CaraesNaur** [comment]
I've worked with several i18n systems, and built a couple myself.

One of the problems with using English as the base language is that its words often have more meanings and nuance than in other languages.

Linguistic concerns don't seem to fit most of these.  For example...

Having a generic "to" is one thing, having additional "to" strings for each of four specific date range pickers doesn't.

I just checked the "to" strings in all the language files.  The only ones where the four datepicker strings differ at all are SK, SR, and SR_lat.  The different one is consistently `selectduedateto`, which is the first line of these in each file... raising the possibility that the other three datepickers were added later and wrongly inherited the generic "to" value.

Each of the four only appear in the task search form in calls to `tpl_datepicker()`.

Surprisingly, generic "to" only appears in the reports search form, also in a call to `tpl_datepicker`().

All five currently exist in functionally identical contexts (the datepicker), and could be combined.  However, it would be wise to keep the generic "to" separate for other purposes and combine the remaining four into something like `datepickerto`.

The same is not the case with "from", as those corresponding strings in English explicitly name the specific datepickers.  The different "to" strings may have been created as excessively direct correlations.
